### PR TITLE
Group DFC API operations by endpoint in documentation

### LINKED ISSUE #11464: [Flaky] ./engines/dfc_provider/spec/requests/addresses_spec.rb
### What we should change and why (this is tech debt)

```

An error occurred while loading ./engines/dfc_provider/spec/requests/addresses_spec.rb.
Failure/Error: require DfcProvider::Engine.root.join("spec/swagger_helper")

NameError:
  uninitialized constant DfcProvider::Engine

  require DfcProvider::Engine.root.join("spec/swagger_helper")
                     ^^^^^^^^
# ./engines/dfc_provider/spec/requests/addresses_spec.rb:3:in `<top (required)>'
No examples found.


Finished in 0.00005 seconds (files took 0.1652 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples
```

### Context

This came up here:
https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/5984505939/job/16237492595

See also Slack thread:
https://openfoodnetwork.slack.com/archives/C2GQ45KNU/p1693209600572129

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

- Closes  #11464

Preparing the addition of more API endpoints.

#### What should we test?

- Visit `/api-docs`
- You see V1 and DFC APIs documented.
- Both APIs have headings for each resource e.g. Customers, Enterprises

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1 or DFC)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by rioug — COMMENTED** [review]
Nice one :+1:

**Review by rioug — APPROVED** [review]

File: spec/base_spec_helper.rb [hunk 1]
```diff
@@ -102,8 +102,12 @@
   # Take example responses from Rswag specs for API documentation.
   # https://github.com/rswag/rswag#enable-auto-generation-examples-from-responses
   config.after(:each, :rswag_autodoc) do |example|
+    # Categories and group operations of the same API endpoint.
+    example.metadata[:operation][:tags] ||= [self.class.top_level_description]
```
Comment (dacook) [inline spec/base_spec_helper.rb:106]: ✨

**Review by dacook — APPROVED** [review]
Great tidy up, and I like the automated categorising!

**Comment by dacook** [comment]
Looks good! ✅ 

Screen Shot 2023-09-13 at 2 55 22 pm

(Although I think the swagger styles could do with a designer's touch..)

**Pull request merged**